src/ResourceManager/Compute/Commands.Compute/Strategies/Compute/ImageEx.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Microsoft.Azure.Commands.Common.Strategies;
+using Microsoft.Azure.Commands.Common.Strategies.Compute;
+using Microsoft.Azure.Management.Compute;
+using Microsoft.Azure.Management.Compute.Models;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Commands.Compute.Strategies.Compute
+{
+    static class ImageEx
+    {
+        public static async Task<ImageAndOsType> UpdateImageAndOsTypeAsync(
+            this IClient client, string imageName, string location)
+        {
+            if (imageName.Contains(':'))
+            {
+                var imageArray = imageName.Split(':');
+                if (imageArray.Length != 4)
+                {
+                    throw new InvalidOperationException("Invalid ImageName");
```

Comment:
Place in a resource string, also, please include the invalid image name so it is clear what the exact failure is (this can be important if it is executed from a script)